azurerm/resource_arm_eventhub_namespace_disaster_recovery_config.go

```diff
@@ -0,0 +1,319 @@
+package azurerm
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/eventhub/mgmt/2017-04-01/eventhub"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmEventHubNamespaceDisasterRecoveryConfig() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmEventHubNamespaceDisasterRecoveryConfigCreate,
+		Read:   resourceArmEventHubNamespaceDisasterRecoveryConfigRead,
+		Update: resourceArmEventHubNamespaceDisasterRecoveryConfigUpdate,
+		Delete: resourceArmEventHubNamespaceDisasterRecoveryConfigDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateEventHubAuthorizationRuleName(),
+			},
+
+			"namespace_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateEventHubNamespaceName(),
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"partner_namespace_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"alternate_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: azure.ValidateEventHubNamespaceName(),
+			},
+
+			"wait_for_replication": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"wait_for_name_to_be_available_on_deletion": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
```

Comment:
can we remove this field? the Delete method should wait until this is fully removed